src/domain/repositories/ProgramsRepository.ts

```diff
@@ -1,20 +1,47 @@
-import { Async } from "domain/entities/Async";
-import { Id } from "domain/entities/Base";
-import { Pair } from "scripts/common";
+import { Enrollment, Event, TrackedEntity } from "domain/entities/TrackedEntity";
+import { CancelableResponse } from "types/d2-api";
 
 export interface ProgramsRepository {
-    closePatients(options: ClosePatientsOptions): Async<void>;
+    get(options: GetOptions): CancelableResponse<ApiGetResponse>;
+    save(payload: Payload): CancelableResponse<ApiSaveResponse>;
 }
 
-export interface ClosePatientsOptions {
-    orgUnitsIds?: Id[];
+export interface GetOptions {
+    programId: string;
+    orgUnitsIds?: string[];
     startDate?: string;
     endDate?: string;
-    programId: Id;
-    programStagesIds: Id[];
-    closureProgramId: Id;
-    timeOfReference: string;
-    pairsDeValue: Pair[];
-    comments?: Pair;
-    post: boolean;
+}
+
+export interface Payload {
+    enrollments: Enrollment[];
+    events: Event[];
+}
+
+export interface ApiGetResponse {
+    instances: TrackedEntity[];
+}
+
+export interface ApiSaveResponse {
+    validationReport?: { errorReports?: Report[] };
+    status: "OK" | "ERROR" | "WARNING";
+    stats: Stats;
+}
+
+export type ApiPostErrorResponse = {
+    response: {
+        data: ApiSaveResponse & { message?: string };
+    };
+};
```

Comment:
Same for these two interfaces. If you need some of that info in the use case, create a specific type containing only the things you need (stats?). Implementation details in data repositories (dhis2 api/ d2-api in this case) should not leak into the domain.